### ISSUE TITLE
shufflejoin and ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by antonkulaga** [bigdatagenomics/adam#1436]
I am trying to reproduce bedtools operations with adam and I often get annoying java.lang.ArrayIndexOutOfBoundsException errors on shuffleJoin for this data ( http://quinlanlab.org/tutorials/bedtools/bedtools.html ). In particular I am doing shufflejoiin for 
cpg.bed and exons.bed
```
org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 15.0 failed 1 times, most recent failure: Lost task 0.0 in stage 15.0 (TID 111, localhost): java.lang.ArrayIndexOutOfBoundsException: 1484
```

**Comment by fnothaft** [comment]
OOC, have you used the new implementation in #1324? It is a really clean rewrite of the ShuffleRegionJoin core. CC @devin-petersohn

**Comment by antonkulaga** [comment]
@fnothaft I am using latest stable ADAM release, there broadcastJoining freezes (even though the bed files are supersmall as you may see) while shuffleJoin gives this annoying java.lang.ArrayIndexOutOfBoundsException.

**Comment by antonkulaga** [comment]
Hm, I think it is spark-notebook to blame for broadcastJoining freezing, after reload it was ok.
By the way, in what situations do you recommend shuffleJoining and in what broadcastJoining?

**Comment by devin-petersohn** [comment]
Hi @antonkulaga. I am trying to reproduce the `ArrayIndexOutOfBoundsException` but I am having a hard time doing so with the files you linked to. I attempted all types of shuffleRegionJoins joining both datasets and had no issues. Is there something specific you were doing when this happened?

**Comment by antonkulaga** [comment]
@devin-petersohn I have not tried your PR, maybe in your PR everything works fine

**Comment by heuermh** [comment]
@antonkulaga Can this issue be closed?

**Comment by devin-petersohn** [comment]
Resolved with #1324
